src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -11155,7 +11155,22 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
 
         let supportEnforceAttributeTargets =
             (g.langVersion.SupportsFeature(LanguageFeature.EnforceAttributeTargets) && memberFlagsOpt.IsNone && not attrs.IsEmpty)
-            && not isVolatile // // VolatileFieldAttribute has a special treatment(specific error FS823)
+            && not isVolatile // VolatileFieldAttribute has a special treatment(specific error FS823)
+
+        // Check for attributes in unit-of-measure expressions
+        // e.g. let x = 1.0<m>
+        //                  ^
+        let rec checkAttributeInMeasure ty =
+            match stripTyEqns g ty with
+            | TType_app(typeInstantiation= [ TType_measure tm ]) -> CheckUnitOfMeasureAttributes g tm
```

Comment:
What if the expression has more than 1 type parameter?

Like an expression creating a new object `Dictionary<int<kg>,int<cm>>` or similar.
If I read this correctly, this only handles applications with exactly 1 type argument.

Reply:
This is fixed by moving the check `TcConstExpr`.